runners/prism/java/src/test/java/org/apache/beam/runners/prism/PrismLocatorTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.prism;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.apache.beam.runners.prism.PrismRunnerTest.getLocalPrismBuildOrIgnoreTest;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link PrismLocator}. */
+@RunWith(JUnit4.class)
+public class PrismLocatorTest {
+
+  @Test
+  public void givenVersionOverride_thenResolves() throws IOException {
+    PrismPipelineOptions options = options();
+    options.setPrismVersionOverride("2.57.0");
+    PrismLocator underTest = new PrismLocator(options);
+    String got = underTest.resolve();
+    assertThat(got).contains(".apache_beam/cache/prism/bin/");
+    assertThat(got).contains("2.57.0");
+    Path gotPath = Paths.get(got);
+    assertThat(Files.exists(gotPath)).isTrue();
+    Files.delete(gotPath);
+  }
+
+  @Test
+  public void givenHttpPrismLocationOption_thenResolves() throws IOException {
+    PrismPipelineOptions options = options();
+    options.setPrismLocation(
+        "https://github.com/apache/beam/releases/download/v2.57.0/apache_beam-v2.57.0-prism-darwin-arm64.zip");
+    PrismLocator underTest = new PrismLocator(options);
+    String got = underTest.resolve();
+    assertThat(got).contains(".apache_beam/cache/prism/bin/");
+    Path gotPath = Paths.get(got);
+    assertThat(Files.exists(gotPath)).isTrue();
+    Files.delete(gotPath);
+  }
+
+  @Test
+  public void givenFilePrismLocationOption_thenResolves() throws IOException {
```

Comment:
I don't think this actually effectively tests that we're pulling the local prism runner. It just tests that we're getting something, but it could be downloaded, right? Or even cached? Could we instead:
a) clear the cache (probably this should be a setup step for each test)
b) write a local temp file and give that as the location
c) compare that the files are identical

Reply:
Note that deleting the file at the end of the test doesn't necessarily resolve the caching issue since a different version of the file could be there if:
1) This is the first test to run and another codeblock populated the cache
2) A previous test failed in the middle and didn't delete the file

So it is better to delete it at the start if it exists (and still do cleanup at the end)

Reply:
So the change you are suggesting is to add test setup to delete any files? A local temp file is out of scope because the goal of PrismLocator is to place the file in a prescribed location to maintain consistency with the Python prism runner.

Reply:
Added a `@Before` method in the test to delete the files under test. Could you please let me know if I missed anythiing else in your two comments here?

Reply:
FYI temp would require setting the destination which is not done to keep consistent with the Python prism runner. The behavior under test is both the source and destination. Only the source is configurable. Yes, there could be a hidden configuration for the destination but I wanted to keep consistent with what the Python prism runner does i.e. constant `~/.apache_beam/...`

Reply:
This LGTM now, thanks!